applications/ThresholdSignature.md

```diff
@@ -0,0 +1,300 @@
+    fn compute_challenge(R: &GroupCommitment, verifying_key: &VerifyingKey, msg: &[u8]) -> Challenge;
+
+    /// Generates the signing nonces and commitments to be used in the signing operation.
+    ///
+    /// Implements [`commit`] from the spec: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-15.html#name-round-one-commitment
+    pub fn commit(secret: &SigningShare) -> (SigningNonces, SigningCommitments);
+
+    /// Receives the message to be signed and a set of signing commitments and a set of randomizing commitments to be used in that signing operation, including that for this participant.
+    ///
+    /// Implements [`sign`] from the spec: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-15.html#name-round-two-signature-share-g
+    pub fn sign(signing_package: &SigningPackage, signer_nonces: &SigningNonces, key_package: &KeyPackage) -> Result<SignatureShare, Error>;
+
+    /// Aggregates the signature shares to produce a final signature that can be verified with the group public key.
+    ///
+    /// Implements [`aggregate`] from the spec: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-15.html#name-signature-share-aggregation
+    pub fn aggregate(signing_package: &SigningPackage, signature_shares: &BTreeMap<Identifier, SignatureShare>, pubkeys: &PublicKeyPackage) -> Result<Signature, Error>;
+
+    /// Verifies if a signature share issued by a participant is valid before aggregating it into a final joint signature to publish.
+    ///
+    /// Implements [`verify_signature_share`] from the spec: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-15.html#name-signature-share-verification
+    pub(crate) fn verify_signature_share(&self, identifier: Identifier, group_commitment_share: &GroupCommitmentShare, verifying_share: &VerifyingShare, lambda_i: Scalar, challenge: &Challenge) -> Result<(), Error>;
+
+    /// An alternative to [`commit`] (not part of the spec) that can be used to pre-process more than 1 nonce for future 1-round FROST signings instead of regular 2-round FROST.
+    pub fn preprocess(num_nonces: u8, secret: &SigningShare) -> (Vec<SigningNonces>, Vec<SigningCommitments>);
+}
+```
+
+#### SimplPedPop Module (Rust)
+
+As of note, an implementation is currently being developed based on [SimplPedPop](https://eprint.iacr.org/2023/899), called [ChillDKG](https://github.com/BlockstreamResearch/bip-frost-dkg), by one of the authors of the paper.
+
+```rust
+pub mod simplpedpop {
+    /// Generates the SimplPedPop messages to be sent to the coordinator or directly to the other participants.
+    pub fn generate_simplpedpop(identifier: Identifier, max_signers: u16, min_signers: u16) -> Result<(SecretPackage, Package), Error>;
+
+    /// Verifies the messages received from all participants and returns the data for the equality check and DKG output.
+    pub fn verify_simplpedpop(secret_package: &SecretPackage, packages: &BTreeMap<Identifier, Package) -> Result<(KeyPackage, PublicKeyPackage), Error>;
```

Comment:
I suppose `SecretPackage` could be a session type too, but maybe not so important.